.evergreen/install-dependencies.sh

```diff
@@ -13,7 +13,7 @@ export SWIFTENV_ROOT="${INSTALL_DIR}/swiftenv"
 export PATH=/opt/cmake/bin:${SWIFTENV_ROOT}/bin:$PATH
 
 # should be set by EVG eventuallty
-LIBMONGOC_VERSION="r1.13"
+LIBMONGOC_VERSION="r1.14"
```

Comment:
1.13 does not work with OpenSSL 1.1.1 which is what is on the Ubuntu 18.04 evg hosts. I guess it's maybe not ideal to not be testing against our min required version, but we do still use 1.13 on Travis anyway, and are going to drop 1.13 support pretty soon, so I'm not too concerned.